Sprint 13

Skoch33 opened the pull request:
1. Проведён рефакторинг WebViewViewController
2. WebViewController покрыт Unit-тестами
3. Написан UI тест, проверяющий авторизацию в приложение
4. Написан UI тест, проверяющий работу с лентой
5. Написан UI тест, проверяющий работу с профилем
6. Приложение поддерживает устройства iPhone с iOS 13 или выше, предусмотрен только портретный режим

Skoch33 committed:
Провел рефакторинг и создал для каждой ответственности свою сущность в коде
Код готов к тестам

Skoch33 committed:
Написал Unit-тесты для WebView

Skoch33 committed:
Рефакторинг ProfileView и тесты к нему

Skoch33 committed:
Рефакторинг ImageList и unit тесты к нему

Skoch33 committed:
Написал UI тесты

amardanov commented on ImageFeed/Auth/AuthViewController.swift:
```diff
@@ -22,10 +22,14 @@ final class AuthViewController: UIViewController {
                 let webViewViewController = segue.destination as? WebViewViewController
-            else { assert(false)
-                assertionFailure("Failed to prepare for \(ShowWebViewSegueIdentifier)")
+            else { assertionFailure("Failed to prepare for \(ShowWebViewSegueIdentifier)")
```
Конструкции `assertionFailure` нужны только для тестов, лучше убирать их из кода

amardanov commented on ImageFeed/ImagesList/ImagesListPresenter.swift:
```diff
@@ -0,0 +1,68 @@
+        })
+    }
+
+    internal func makeAlert(with error: Error) -> UIAlertController {
```
internal можно не писать, все функции по умолчанию internal

amardanov commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -67,12 +69,13 @@ final class ImagesListViewController: UIViewController {
     
     //MARK: - Private Properties
-    
+    lazy var presenter: ImagesListPresenterProtocol? = {
```
обычно используем такой порядок: сначала перечисляем свойства класса (сначала UI, потом сервисы и прочие свойства), затем функции. 
  При этом также делим на приватные и публичные блоки. 
  Функции перечисляют в таком порядке: `init`ы, `override`ы, блоки публичных и приватных функций, и далее блок c аннотацией `@IBAction`

amardanov commented on ImageFeed/Profile/ProfileService.swift:
```diff
@@ -6,63 +6,71 @@
-}
+
+     func fetchProfile(_ token: String, completion: @escaping (Result<Profile, Error>) -> Void) {
+         assert(Thread.isMainThread)
```
также про ассерт

amardanov commented on ImageFeed/Profile/ProfileService.swift:
```diff
@@ -6,63 +6,71 @@
+     }
+ }
+
+ struct ProfileResult: Codable {
```
вынести в отдельные файлы каждую структуру

amardanov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -8,175 +8,140 @@
+
+final class ProfileViewController: UIViewController, ProfileViewControllerProtocol {
+
+    lazy var presenter: ProfilePresenterProtocol = {
```
все UI свойства private

amardanov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -8,175 +8,140 @@
-        present(alert, animated: true, completion: nil)
+}
+
+// MARK: - Extensions
```
// MARK: - Extensions не нужен тут, одного хватит

Skoch33 committed:
1. Поправил код, приложение запускается
2. Вынес реализацию метода clean() из WebViewViewController
3. UI тесты проходят
4. Убрал конструкции assertionFailure из. кода
5. Вынес в отдельные файлы каждую структуру в ProfileService

Skoch33 committed:
Поправил код и тесты, приложение стало билдиться намного быстрее

Skoch33 committed:
Поправил UI тест testAuth() , клавиатура убирается, ввод логина и пароля происходит без проблем

Skoch33 committed:
Правки по ревью

Skoch33 committed:
Последняя правка
